File: CS-IOS/Classes/Core/Categories/UIKit/UIScreen+CSExtension.m
//
// Created by Rene Dohan on 19/02/18.
// Copyright (c) 2018 renetik_software. All rights reserved.
//

#import "UIScreen+CSExtension.h"
#import "CSLang.h"

@implementation UIScreen (CSExtension)

+ (BOOL)isPortrait {
	let window = UIApplication.sharedApplication.delegate.window;
	if(window.rootViewController) {
		let orientation =
		window.rootViewController.interfaceOrientation;
		return UIInterfaceOrientationIsPortrait(orientation);
	} else {
		let orientation =
		UIApplication.sharedApplication.statusBarOrientation;
		return UIInterfaceOrientationIsPortrait(orientation);
	}
}

+ (BOOL)isLandscape {
	return !self.isPortrait;
}

+ (CGFloat)height {
	return UIScreen.mainScreen.bounds.size.height;
}

+ (CGFloat)width {
	return UIScreen.mainScreen.bounds.size.width;
}
@end

File: Renetik/Renetik/Classes/Core/Extensions/UIKit/UIScreen+CSExtension.swift
//
//  UIScreen.swift
//  Renetik
//
//  Created by Rene Dohan on 8/24/19.
//

import Foundation
import UIKit

public extension UIScreen {
    public class var size: CGSize { main.bounds.size }
    public class var width: CGFloat { size.width }
    public class var height: CGFloat { size.height }
    public class var maxLength: CGFloat { max(width, height) }
    public class var minLength: CGFloat { min(width, height) }
    public class var isZoomed: Bool { main.nativeScale >= main.scale }
    public class var isRetina: Bool { main.scale >= 2.0 }
    public class var orientation: UIInterfaceOrientation {
        UIApplication.shared.delegate?.window??.rootViewController?.interfaceOrientation ??
            UIApplication.shared.statusBarOrientation
    }
    public class var isPortrait: Bool { orientation.isPortrait }
    public class var isLandscape: Bool { !isPortrait }
    public class var isLandscapeLeft: Bool { orientation == .landscapeLeft }
    public class var isLandscapeRight: Bool { orientation == .landscapeRight }
    public class var isThin: Bool { isPortrait && UIDevice.isPhone }
    public class var isUltraThin: Bool { isPortrait && UIScreen.width <= 375 }
    public class var isWide: Bool { !isThin }
    public class var isUltraWide: Bool { UIDevice.isTablet && isLandscape }
    public class var isShort: Bool { isLandscape && UIDevice.isPhone }
    public class var isUltraShort: Bool { isLandscape && isSmallPhone }
    public class var isTall: Bool { !isShort } // tablet || portrait
    public class var isUltraTall: Bool { UIDevice.isTablet || !isSmallPhone && isPortrait }
    public class var isSmallPhone: Bool {
        (UIDevice.typeIsLike == .iphone4 || UIDevice.typeIsLike == .iphone5)
    }
}
